lib/Mrm/Mrmwcrw.c: warning: dereferencing type-punned pointer will break strict-aliasing rules

During compilation of `Mrmwcrw.c`:
```
/bin/sh ../../libtool  --tag=CC   --mode=compile x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I/tmp/portage/x11-libs/motif-2.4.0/work/motif-2.4.0/lib/Mrm -I../../include -I../../lib/Xm  -I/tmp/portage/x11-libs/motif-2.4.0/work/motif-2.4.0/lib/Mrm/.. -I.. -I../Xm -I/usr/include/uuid -I/usr/include/freetype2 -DNDEBUG  -march=native -ggdb -O2 -pipe -Wall -Wshadow -Wwrite-strings -Wcast-align -Wformat-security -Wno-unused -Wno-unused-result -Wno-misleading-indentation -Wno-implicit-fallthrough -Wno-int-in-bool-context -Wno-conversion -Wno-discarded-qualifiers -Wno-cast-qual -c -o Mrmwcrw.lo /tmp/portage/x11-libs/motif-2.4.0/work/motif-2.4.0/lib/Mrm/Mrmwcrw.c
libtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I/tmp/portage/x11-libs/motif-2.4.0/work/motif-2.4.0/lib/Mrm -I../../include -I../../lib/Xm -I/tmp/portage/x11-libs/motif-2.4.0/work/motif-2.4.0/lib/Mrm/.. -I.. -I../Xm -I/usr/include/uuid -I/usr/include/freetype2 -DNDEBUG -march=native -ggdb -O2 -pipe -Wall -Wshadow -Wwrite-strings -Wcast-align -Wformat-security -Wno-unused -Wno-unused-result -Wno-misleading-indentation -Wno-implicit-fallthrough -Wno-int-in-bool-context -Wno-conversion -Wno-discarded-qualifiers -Wno-cast-qual -c /tmp/portage/x11-libs/motif-2.4.0/work/motif-2.4.0/lib/Mrm/Mrmwcrw.c  -fPIC -DPIC -o .libs/Mrmwcrw.o
/tmp/portage/x11-libs/motif-2.4.0/work/motif-2.4.0/lib/Mrm/Mrmwcrw.c: In function 'Urm__CW_ConvertValue':
/tmp/portage/x11-libs/motif-2.4.0/work/motif-2.4.0/lib/Mrm/Mrmwcrw.c:2483:31: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
 2483 |         float_val   = (float)*(double *)unitsfloatvalue->value;
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This triggers a QA warning in Gentoo Linux:
```
 * QA Notice: Package triggers severe warnings which indicate that it
 *            may exhibit random runtime failures.
 * /tmp/portage/x11-libs/motif-2.4.0/work/motif-2.4.0/lib/Mrm/Mrmwcrw.c:2483:31: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]

 * Please do not file a Gentoo bug and instead report the above QA issues directly
 * to the upstream developers of this software.
```

--- a/lib/Mrm/Mrmwcrw.c
+++ b/lib/Mrm/Mrmwcrw.c
@@ -2480,7 +2480,8 @@ Urm__CW_ConvertValue (Widget			parent,
 	screen = DefaultScreenOfDisplay(display);
 	unitsfloatvalue = (RGMUnitsFloatPtr) *val;
 	float_units = unitsfloatvalue->units;
-	float_val   = (float)*(double *)unitsfloatvalue->value;
+	memcpy(&double_val, unitsfloatvalue->value, sizeof double_val);
+	float_val = (float)double_val;
 
 	if (float_val != 0)
 	  {

--- a/m4/ax_cflags.m4
+++ b/m4/ax_cflags.m4
@@ -29,6 +29,7 @@ AC_DEFUN([AX_CFLAGS],[
 		ax_cc_gcov_command="llvm-cov gcov"
 		AX_APPEND_COMPILE_FLAGS([ dnl
 			-Wall dnl
+			-Wstrict-aliasing dnl
 			-Wno-unused dnl
 			-Wno-unused-result dnl
 			-Wno-misleading-indentation dnl
@@ -45,6 +46,7 @@ AC_DEFUN([AX_CFLAGS],[
 			-Wall dnl
 			-Wshadow dnl
 			-Wwrite-strings dnl
+			-Wstrict-aliasing dnl
 			-Wcast-align dnl
 			-Wformat-security dnl
 			-Wno-unused dnl

